parser/src/powdr.lalrpop

```diff
@@ -681,6 +683,25 @@ PatternList: Vec<Pattern> = {
 
 // ---------------------------- Trait/Impl Declaration -----------------------------
 
+GenericTraitName: (String, Option<TraitScheme<Expression>>) = {
+    <name:Identifier> => (name, None),
```

Comment:
If we require traits to have at least one type var, then it doesn't make sense to parse impls without at least type arguments for the trait (it is legal to not declare any type params like in `impl Into<StructA, StructB> { ... }`

Reply:
Oh and `name` should actually be a Path here, because it is a reference and not a new declaration.